include/fuse.h

```diff
@@ -750,6 +750,19 @@ struct fuse_operations {
 	 */
 	int (*fallocate) (const char *, int, off_t, off_t,
 			  struct fuse_file_info *);
+
+	/**
+	 * Copy a range of data from one file to another
+	 *
+	 * Performs an optimized copy between two file descriptors without the
+	 * additional cost of transferring data through the FUSE kernel module
+	 * to user space (glibc) and then back into the FUSE filesystem again.
```

Comment:
Please also describe what happes if this method isn't implemented (I assume the kernel will assume a bunch of separate read() and write() requests)?

Reply:
Currently it is actually glibc doing a fallback. There seems to be work in progress in the kernel to do this, but that has not been merged yet (or even no final design?).